music_assistant/server/providers/bluesound/__init__.py

```diff
@@ -0,0 +1,586 @@
+
+        if not self.mass_player:
+            return
+        if self.sync_status.volume == -1:
+            self.mass_player.volume_level = 100
+        else:
+            self.mass_player.volume_level = self.sync_status.volume
+        self.mass_player.volume_muted = self.status.mute
+
+        # self.logger.debug(self.status.input_id)
+        if self.status.state == "stream" and self.status.input_id == "input0":
+            self.mass_player.active_source = SOURCE_LINE_IN
+        elif self.status.state == "stream" and self.status.input_id == "Airplay":
+            self.mass_player.active_source = SOURCE_AIRPLAY
+        elif self.status.state == "stream" and self.status.input_id == "Spotify":
+            self.mass_player.active_source = SOURCE_SPOTIFY
+        elif self.status.state == "stream" and self.status.input_id == "RadioParadise":
+            self.mass_player.active_source = SOURCE_RADIO
+
+        # Music Assistant falls in this category
+        # elif self.status.state ==  "stream" and self.status.input_id == "None":
+        #    self.mass_player.active_source = SOURCE_UNKNOWN
+
+        # TODO check pair status
+
+        # TODO fix pairing
+
+        if self.sync_status.master is None:
+            if self.sync_status.slaves:
+                self.mass_player.group_childs = (
+                    self.sync_status.slaves if len(self.sync_status.slaves) > 1 else set()
+                )
+                self.mass_player.synced_to = None
+
+            if self.status.state == "stream":
+                self.mass_player.current_media = PlayerMedia(
+                    uri=self.status.stream_url,
+                    title=self.status.name,
+                    artist=self.status.artist,
+                    album=self.status.album,
+                    image_url=self.status.image,
+                    # Does not make sense as bluesound does not report song length
+                    # duration=self.status.total_seconds / 1000,
+                )
+
+            # TODO fix sync and multiple players
+
+            # elif container.get("name") and container.get("id"):
+            #     images = self.status.image
+            #     image_url = self.status.image
+            #     self.mass_player.current_media = PlayerMedia(
+            #         uri=self.status.stream_url,
+            #         title=self.status.name,
+            #         image_url=self.status.image,
+            #     )
+            else:
+                self.mass_player.current_media = None
+
+        else:
+            self.mass_player.group_childs = set()
+            self.mass_player.synced_to = self.sync_status.master
+            self.mass_player.active_source = self.sync_status.master
+
+        # self.mass_player.state = PLAYBACK_STATE_MAP[active_group.playback_state]
+
+        self.mass_player.elapsed_time = self.status.seconds
+        self.mass_player.elapsed_time_last_updated = time.time()
+        self.mass.players.update(self.player_id)
```

Comment:
is this side effect wanted here ? And if desired here then please move it to the last line